modulus/experimental/resdiff/training/dataset.py

```diff
@@ -283,10 +282,11 @@ def _load_raw_labels(self):
 
 #Era5
 class Era5Dataset(torch.utils.data.Dataset):
-  def __init__(self, params, path, train, cache=True, task='sr'):
+  def __init__(self, params, path, train, dist, cache=True, task='sr'):
```

Comment:
For constructors I would definitely suggest just doing instead of argument.

self.dist = DistributedManager()